error.go

```diff
@@ -8,11 +8,18 @@ import (
 
 // APIError provides error information returned by the OpenAI API.
 type APIError struct {
-	Code           any     `json:"code,omitempty"`
-	Message        string  `json:"message"`
-	Param          *string `json:"param,omitempty"`
-	Type           string  `json:"type"`
-	HTTPStatusCode int     `json:"-"`
+	Code           any        `json:"code,omitempty"`
+	Message        string     `json:"message"`
+	Param          *string    `json:"param,omitempty"`
+	Type           string     `json:"type"`
+	HTTPStatusCode int        `json:"-"`
+	InnerError     InnerError `json:"innererror,omitempty"`
+}
+
+// InnerError Azure Content filtering.
+type InnerError struct {
```

Comment:
@ZeroDeng01 Please explain in the comments that InnerError struct is only valid for Azure OpenAI Service.

Reply:
Complete